src/crypto/holder.rs

```diff
@@ -0,0 +1,50 @@
+use once_cell::sync::OnceCell;
+use failure::Fail;
+use super::Cryptographer;
+
+static CRYPTOGRAPHER: OnceCell<&'static dyn Cryptographer> = OnceCell::INIT;
+
+#[derive(Debug, Fail)]
+#[fail(display = "Cryptographer already initialized")]
+pub struct SetCryptographerError(());
+
+/// Sets the global object that will be used for cryptographic operations.
+///
+/// This is a convenience wrapper over [`set_cryptographer`],
+/// but takes a `Box<dyn Cryptographer>` instead.
+#[cfg(not(any(feature = "use_ring", feature = "use_openssl")))]
+pub fn set_boxed_cryptographer(c: Box<dyn Cryptographer>) -> Result<(), SetCryptographerError> {
+    set_cryptographer(Box::leak(c))
```

Comment:
Maybe add a comment why this is OK to leak the `Box` as I'm just trusting you know your rust-foo enough to convince me this is fine.